datafusion/sql/src/statement.rs

```diff
@@ -661,27 +661,35 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             .get_table_provider((&table_name).into())?;
         let arrow_schema = (*provider.schema()).clone();
         let table_schema = Arc::new(DFSchema::try_from(arrow_schema)?);
-        let mut values: BTreeMap<_, _> = table_schema
-            .fields()
-            .iter()
-            .map(|f| {
-                (
-                    f.name().clone(),
-                    ast::Expr::Identifier(ast::Ident::from(f.name().as_str())),
-                )
-            })
-            .collect();
+        let values = table_schema.fields().iter().map(|f| {
+            (
+                f.name().clone(),
+                ast::Expr::Identifier(ast::Ident::from(f.name().as_str())),
+            )
+        });
 
         // Overwrite with assignment expressions
         let mut planner_context = PlannerContext::new();
-        for assign in assignments.iter() {
-            let col_name: &Ident = assign
-                .id
-                .iter()
-                .last()
-                .ok_or(DataFusionError::Plan("Empty column id".to_string()))?;
-            let _ = values.insert(col_name.value.clone(), assign.value.clone());
-        }
+        let assign_map = assignments
+            .iter()
+            .map(|assign| {
+                let col_name: &Ident = assign
```

Comment:
I know this PR didn't add it, but the use of `last` seems somewhat wrong here -- maybe  https://github.com/apache/arrow-datafusion/blob/cb05490df1ed27cd90d904296b7866187c6e737b/datafusion/sql/src/planner.rs#L333 or something would work 🤔

Reply:
This is actually looking for the column name - we know the table name already. Is there an equivalent for columns?

Meta-question: is there a method for discoverability of these functions? I often assume there is a helper out there somewhere but struggle to find it...